backend/GameManager.py

```diff
@@ -58,5 +82,173 @@ def get_pieces_left(self):
+
+        # Perform the move
+        self.board.set_position(target_row, target_col, self.turn.name.lower())
+        self.board.set_position(start_row, start_col, Color.EMPTY)
+        # self.board.board[start_row][start_col] = None
+
+        # removing the new position from open move and adding the previous position to it
+        self.open_moves.append((start_row, start_col))
+        self.open_moves.remove((target_row, target_col))
+
+        self.selected_piece = None
+        return True
+
+    def get_piece_at(self, row, col):
+        return self.board.board[row][col]  
+    
+    # def is_mill_formed(self, row, col, turn):
+    #     for mill in self.mills:
+    #         if (row, col) in mill:
+    #             # Check if all three positions in this mill belong to the current player
+    #             first = self.get_piece_at(r, c)
+                
+    #             if all(self.get_piece_at(r, c) == turn for r, c in mill):
+    #                 return True
+    #     return False
+
+
+    def is_mill_formed(self, row, col):
+        for mill in self.mills:
+            if (row, col) in mill:
+                # Create an empty list to store comparison results
+                pieces_in_mill = []
+                
+                # Loop through each position in the mill
+                for r, c in mill:
+                    piece_at_position = self.get_piece_at(r, c)  # Get the piece at the mill position
+                    is_current_turn_piece = (piece_at_position.get_state().name == self.get_turn().name)  # Check if it's the current player's turn
+                    pieces_in_mill.append(is_current_turn_piece)  # Append the result (True/False) to the list
+                
+                # Now use 'all' to check if all the values in the list are True
+                if all(pieces_in_mill):
+                    return True
+
+        return False
+    
+    def remove_piece_mill(self, row, col):
+        self.open_moves.append((row, col))
+        self.board.set_position(row, col, Color.EMPTY)
+        self.waiting_for_removal = False  # Reset the removal state
+        self.removable_pieces = []  # Clear the list of removable pieces
+        self.end_turn()  # End the turn after removal
+        self.end_turn()  # swap the turn to opponent
+        return True
+    
+    def remove_opponent_piece(self , pieces_on_board):
+        opponent_turn = Color.BLACK if self.get_turn() == Color.WHITE else Color.WHITE
+        
+        # Get all opponent's pieces
+        opponent_pieces = [
+            (row, col) for (row, col), piece_turn in pieces_on_board.items()
+            if piece_turn == opponent_turn
+        ]
+        
+        # Try to remove non-mill pieces first - logic is wrong but it works as user select write piece
+        opponent_pieces_non_mill = [(row, col) for row, col in opponent_pieces if not self.is_mill_formed(row, col)]
```

Comment:
Once we create `findMill `function can reuse it here.

Reply:
i tried that and it not working, 
this can able to capture opponent non mill piece though there is a bug that needs to be fixed into frontend.